### ISSUE TITLE
Android 11 - It is not redirecting on browser and Lyft app

### ISSUE THREAD

**Issue by deepbhavsar** [lyft/lyft-android-sdk#25]
I am checking Lyft button but I found that it is not redirecting on browser Lyft app. It require to upgrade code for intent and deeplinking.

**Comment by earamsey** [comment]
were you able to resolve this issue?

**Comment by mikaaali** [comment]
Additional note on issue observed: 
Taping on Lyft Button is not redirecting on Android 11(API 30) and Android 12(API 31) when `targetSdk` is set to 30

**Comment by makkhay** [comment]
Yeah, deep linking doesn't work when you have targetSDK set to 30 and above. https://www.reddit.com/r/androiddev/comments/ni33zr/anyone_else_experience_completely_broken_deep/

**Comment by mikaaali** [comment]
I was also having issues getting deep links to work when android `targetSDK` is set to `30 and above`.
Still fairly new at android development. Not sure what the best practice is in this case. After some reading. 
We manage to fix it within our own app by adding a `<queries>` element in the `Android manifest`.

I created a branch with updates to the lyft sample project(https://github.com/mikaaali/lyft-android-sdk/tree/defect/package-visibility-in-android#readme). Hopefully, you guys find this helpful.

**Comment by makkhay** [comment]
@mikaaali hmm interesting, just checked your sample project and it looks like only the Pixel devices are impacted ? I tested on my Samsung device and the deeplinking worked just fine, but however, the drop off location data was not retrieved by the Lyft native app. When you login in to the Lyft native app, were you able to see the drop off location data ?

**Comment by mikaaali** [comment]
@makkhay yeah it's interesting. I think both Pixel and Samsung are impacted when `targetSDK is set to 30 and above`.  
Like you said Samsung seems to works halfway, it does launch deep link but without the drop-off location data. 

So far I haven't observed any error once I updated the app by following this guide: https://developer.android.com/training/package-visibility

**Comment by makkhay** [comment]
@mikaaali  One quick question,  you said you were able to fix the issue by adding a <queries> , I am guessing you added Lyft SDK as a lib instead from the gradle ? Because as far as I know, when you import from gradle, you wouldn't be able to make changes to the SDK code. Thanks

**Comment by mikaaali** [comment]
@makkhay the Lyft SDK is added as a dependency in the `build.gradle` file. No code change is needed for Lyft SDK to solve the deep link issue.

**Comment by makkhay** [comment]
@mikaaali  oh thanks, got it working.
